zookeeper-server/src/main/java/org/apache/zookeeper/cli/SyncCommand.java

```diff
@@ -55,14 +57,18 @@ public CliCommand parse(String[] cmdArgs) throws CliParseException {
     @Override
     public boolean exec() throws CliException {
         String path = args[1];
+        CountDownLatch latch = new CountDownLatch(1);
+        final int[] resultCode = new int[1];
         try {
             zk.sync(path, new AsyncCallback.VoidCallback() {
-
                 public void processResult(int rc, String path, Object ctx) {
-                    out.println("Sync returned " + rc);
+                    resultCode[0] = rc;
+                    latch.countDown();
                 }
             }, null);
-        } catch (IllegalArgumentException ex) {
+            latch.await();
```

Comment:
Wouldn't it be safer to wait for some reasonable amount of time (30s) rather than waiting indefinitely?

Reply:
Maybe it can be a parameter with a default of 30secs